src/views/ManageList.jsx

```diff
@@ -1,7 +1,126 @@
-export function ManageList() {
+import { useState } from 'react';
+import { addItem } from '../api/firebase';
+import toast, { Toaster } from 'react-hot-toast';
+
+const purchaseTimelines = {
+	soon: 7,
+	kindOfSoon: 14,
+	notSoon: 30,
+};
+
+export function ManageList({ listPath }) {
+	const [itemName, setItemName] = useState('');
+	const [itemNextPurchaseTimeline, setItemNextPurchaseTimeline] = useState('');
+
+	const handleItemNameTextChange = (e) => {
+		setItemName(e.target.value);
+	};
+
+	const handleNextPurchaseChange = (e) => {
+		setItemNextPurchaseTimeline(e.target.value);
+	};
+
+	const handleSubmit = async (e) => {
+		e.preventDefault();
+
+		const daysUntilNextPurchase = purchaseTimelines[itemNextPurchaseTimeline];
+
+		try {
+			await toast.promise(
+				addItem(listPath, {
+					itemName,
```

Comment:
I left a similar comment on PR #23 , we might want to try to try and validate this input 😄 

https://github.com/the-collab-lab/tcl-77-smart-shopping-list/pull/23/files#r1730100136

Reply:
also as a pointer - if you're doing the same thing multiple times...might be a sign to make it a utility function :)